pkg/booleanpolicy/networkpolicy/networkpolicy_test.go

```diff
@@ -26,48 +26,81 @@ func policy(classificationEnums []storage.NetworkPolicyType) *storage.NetworkPol
 func (suite *NetworkPolicySuite) Test_FilterForDeployment() {
 	cases := map[string]struct {
 		deploymentLabels        map[string]string
+		specPodTemplateLabels   map[string]string
 		netpolSelectors         []map[string]string
 		expectedPoliciesMatched int
 	}{
 		"Match: one NW policy with one label": {
-			deploymentLabels: map[string]string{"app": "central"},
+			deploymentLabels:      map[string]string{"app": "central"},
+			specPodTemplateLabels: map[string]string{"app": "central"},
 			netpolSelectors: []map[string]string{
 				{"app": "central"},
 			},
 			expectedPoliciesMatched: 1,
 		},
 		"Match: one NW policy with two labels": {
-			deploymentLabels: map[string]string{"app": "central", "env": "prod"},
+			deploymentLabels:      map[string]string{"app": "central", "env": "prod"},
+			specPodTemplateLabels: map[string]string{"app": "central", "env": "prod"},
 			netpolSelectors: []map[string]string{
 				{"app": "central", "env": "prod"},
 			},
 			expectedPoliciesMatched: 1,
 		},
 		"One Match: two NW policies with one label": {
-			deploymentLabels: map[string]string{"app": "central", "env": "prod"},
+			deploymentLabels:      map[string]string{"app": "central", "env": "prod"},
+			specPodTemplateLabels: map[string]string{"app": "central", "env": "prod"},
 			netpolSelectors: []map[string]string{
 				{"app": "central"},
 				{"app": "sensor"},
 			},
 			expectedPoliciesMatched: 1,
 		},
 		"Two Matches: two NW policies with one label": {
-			deploymentLabels: map[string]string{"app": "central", "env": "prod"},
+			deploymentLabels:      map[string]string{"app": "central", "env": "prod"},
+			specPodTemplateLabels: map[string]string{"app": "central", "env": "prod"},
 			netpolSelectors: []map[string]string{
 				{"app": "central"},
 				{"env": "prod"},
 			},
 			expectedPoliciesMatched: 2,
 		},
+		"No Match: NW policies shall not match against the deployment label even if 0 pods match": {
+			deploymentLabels:      map[string]string{"app": "central", "env": "prod"},
+			specPodTemplateLabels: map[string]string{"app": "sensor", "env": "dev"},
```

Comment:
_extreme nit:_ I think it's a bit confusing to use `central` as deployment label and `sensor` as pod label. I like to think of unit tests as examples of how things work, and that is not a "realistic" scenario that might make the reader more confused. I still think that dumb names are more helpful here, something like `label=a` vs `label=b`.

Reply:
After reading through the reamining of the tests I think you could use the same labels from test `"One Match: NW policies shall match against the pod labels only"` but just change the netpol selector to `central-deployment` instead.

Reply:
Okay, that makes sense. I will change this and will also do a rebase hoping for the same green CI as now.